server/services/propertyRating.service.js

```diff
@@ -0,0 +1,29 @@
+import PropertyRating from '../models/propertyRating.model';
+import { ErrorHandler } from '../helpers/errorHandler';
+import httpStatus from '../helpers/httpStatus';
+import { getOfferById } from './offer.service';
+import { OFFER_STATUS } from '../helpers/constants';
+
+export const getRatingById = async (id) => PropertyRating.findById(id).select();
+
+export const addPropertyRating = async (ratingInfo) => {
+  const offer = await getOfferById(ratingInfo.offerId);
+
+  if (!offer) {
+    throw new ErrorHandler(httpStatus.NOT_FOUND, 'Invalid offer');
+  }
+
+  if (offer.status !== OFFER_STATUS.RESOLVED) {
+    throw new ErrorHandler(
+      httpStatus.PRECONDITION_FAILED,
+      'You cannot rate a property you do not own',
+    );
+  }
+
+  try {
+    const addedArea = await new PropertyRating(ratingInfo).save();
```

Comment:
Not sure why you used addedArea here 🤷🏽‍♂️ 

```suggestion
    const addedArea = await new PropertyRating(ratingInfo).save();
```